### ISSUE TITLE
Cover block overlay color empty if disabled by theme.json

### ISSUE THREAD

**Issue by MatzeKitt** [WordPress/gutenberg#46148]
### Description

For my theme, the color settings are removed completely:

```json
{
	"version": 2,
	"settings": {
		"color": {
			"custom": false,
			"customDuotone": false,
			"customGradient": false,
			"duotone": [],
			"gradients": [],
			"link": false,
			"palette": [],
			"text": false,
			"background": false,
			"defaultGradients": false,
			"defaultPalette": false
		}
         }
}
```

Nevertheless, the cover block displays the color setting for the overlay color. But clicking on it returns an empty popover, since there is no color. So if there is no color, the color setting for the cover overlay should be omitted completely.

### Step-by-step reproduction instructions

1. Use the `theme.json` above.
2. Create a new post and add a cover block.
3. Add an image to the cover block.
4. See color settings of the cover overlay.

### Screenshots, screen recording, code snippet

<img width="337" alt="Bildschirm­foto 2022-11-29 um 11 41 04" src="https://user-images.githubusercontent.com/1018979/204508657-38ddaef7-a90f-4a8b-a1d8-182c9fac92bb.png">

### Environment info

- WordPress 6.1, no Gutenberg plugin installed

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by t-hamano** [comment]
@MatzeKitt 
Thank you for submitting the issue.
I was able to reproduce the issue, and it seems to occur in all blocks with `ColorGradientSettingsDropdown`.

<details>
<summary>Cover</summary>

![cover](https://user-images.githubusercontent.com/54422211/205474991-17b3a9b5-4aa7-4232-b0c7-6206194f0e0d.png)

</details>

<details>
<summary>Post Featured Image</summary>

![post-featured-image](https://user-images.githubusercontent.com/54422211/205474996-0b76384e-fa0f-46dc-8b83-9060c4246f8a.png)

</details>

<details>
<summary>Social Icons</summary>

![social-icons](https://user-images.githubusercontent.com/54422211/205474998-73e1b67c-de64-4203-8120-49f2d63a30f9.png)

</details>

<details>
<summary>Navigation</summary>

![navigation](https://user-images.githubusercontent.com/54422211/205475000-a96d9403-5329-447a-ad53-514d97187cd6.png)

</details>

**Comment by t-hamano** [comment]
I have discovered that this problem has been resolved by #50115.
It worked as expected on all the blocks I mentioned in [this comment](https://github.com/WordPress/gutenberg/issues/46148#issuecomment-1336325367), not just the cover block.

@MatzeKitt
I would like to close this issue, but if you have any questions, please feel free to comment.
